src/test/java/io/javalin/openapi/TestOpenApiAnnotations_Java.java

```diff
@@ -302,8 +354,22 @@ public void testIfUserIsWarnedOnInvalidPath() {
         });
         assertThat(log).contains(
             "The `path` of one of the @OpenApi annotations on io.javalin.openapi.ClassHandlerWithInvalidPath is incorrect. " +
-            "The path param \":id\" is documented, but couldn't be found in GET \"/account\". " +
-            "Do you mean GET \"/account/:id\"?"
+                "The path param \":id\" is documented, but couldn't be found in GET \"/account\". " +
+                "Do you mean GET \"/account/:id\"?"
         );
     }
+
+    @Test
+    public void testThatSchemaTypeWorks() {
+        OpenApiOptions options = new OpenApiOptions(new Info().title("Example").version("1.0.0"))
+            .activateAnnotationScanningFor("io.javalin.openapi");
+        OpenAPI schema = OpenApiTestUtils.extractSchemaForTest(options, app -> {
+            JavaApiwithAnyOfOneOfReference handler = new JavaApiwithAnyOfOneOfReference();
+            app.get("/anyOfRoute", handler::anyOfMethod);
+            app.get("/oneOfRoute", handler::oneOfMethod);
+            app.get("/allOfRoute", handler::allOfMethod);
+            return Unit.INSTANCE;
+        });
+        OpenApiTestUtils.assertEqualTo(schema, JsonKt.getOneOfAnyOfExampleJson());
+    }
```

Comment:
You just tested the Java Annotations. Could you please also add a test to the `TestOpenApi.kt` file, which generates the same json, but uses the DSL? This is important to make sure that both APIs are supported and easy to use.

Reply:
8e8f01e sort of does this. The cc.vileda dsl does not expose extension-methods to build Mediatype from a Schema (and Schema from a class) or some other thing. The method mediatypeComposed from io.javalin.plugin.openapi.external i used prematurely (with respect to the syntax of the dsl by cc.vileda)